### ISSUE TITLE
Firefox & ESET Endpoint Security 9 - very high CPU usage

### ISSUE THREAD

**Issue by mysteriously** [sandboxie-plus/Sandboxie#1502]
**Describe the problem**
When Firefox 91.4.1 ESR runs in multiple sandboxes, ESET starts using all cpu power. It is not possible to log the user off and shut down the computer.

**To Reproduce**
1. Fun Firefox in first Sandbox
2. Fun Firefox in second Sandbox
3. Close Firefox instance running in one of sandboxes
4. Try to close other Firefox instance in other sandbox. GUI is gone, however Firefox processes will remain running in other sandbox. It is not possible to terminate them even from system account.
5. ESET starts using all cpu power.
6. Force restart the computer as system becomes unresponsive

**Screenshots**
I will try to make screenshots of all errors if I manage to do it in 10 minutes as system works extremely slow when it happens

**Expected behavior**
Just terminate the other Firefox instance.

**System details and installed software**
 - Windows 11 21H2
 - What is your current Sandboxie edition and version? (e.g. Sandboxie Plus 0.9.6 64-bit) - Sandboxie Plus v1.0.6 64-bit
 - What was your previous Sandboxie version before the update, if any? - Sandboxie Plus v1.0.5 64-bit
 - Please mention in which version this bug got introduced, because it acts as a guide for testers and developers. - I don't know as I switched to Firefox very recently
 - Please mention any security software running in the background, even if the real-time protection is disabled. - ESET Endpoint Security 9.0.2032.6

**If you had a compatibility issue or a crash**
 - Please add a download link to the program not working properly in Sandboxie. - https://www.eset.com/int/business/download/endpoint-security-windows/
 - Please consider to take a log and make it available for analysis: https://git.io/Jwj2y - Ok, i will try next time

**Comment by mysteriously** [comment]
https://i.imgur.com/p3tEEc8.png
https://i.imgur.com/BBb2s8u.png
https://i.imgur.com/TnMFq1b.png

**Comment by mysteriously** [comment]
I 'think' I resolved it by exporting ESET configuration, restoring defaults, restoring custom. It started happening after AV build upgrade

**Comment by mysteriously** [comment]
I think it is issue with HIPS

**Comment by isaak654** [comment]
> I think the conflict is in between SBIE and ESET's HIPS

Please test this workaround and report if it helps: https://github.com/sandboxie-plus/Sandboxie/issues/1545#issuecomment-1019023949

**Comment by mysteriously** [comment]
I don't use 'Banking & Payment Protection'. It is already disabled for me

**Comment by 0x391F** [comment]
By the way, sometimes I ran Thunder (XunLei, a popular download tool in China) in Sandboxie, the CPU usage of ekrn.exe was so high that I almost couldn't do anything, exclude restart my computer. But the bug disappeared after restart my computer. 

https://forum.eset.com/topic/30272-ekrnexe-high-cpu-usage-50-due-to-context-switches-frequently/

See the reply on January 31.

**Comment by mysteriously** [comment]
So far no luck. I think it is related to #1514 so I decided to revoke new folders creation for all Sandboxes `%SBIE%\%username%\Firefox\user\current\AppData\Local\Mozilla\Firefox\Profiles\%PROFILE%\cache2`. I will let you know if it worked

**Comment by mysteriously** [comment]
This issue is few years old. Happened with ESET Internet Security 6 and SBIE by Invincea. If there is anyone who want to troubleshoot it with joined forces, you know where to find me. 1 hour is not enough to run process explorer or process monitor when the issue occur which makes the troubleshooting impossible

**Comment by mysteriously** [comment]
Sandboxie crashed when this issue occurred. However due to extreme slowness I was not able to take screenshot showing sandboxie dumpfile location. I can't find it now

**Comment by 0x391F** [comment]
> By the way, sometimes I ran Thunder (XunLei, a popular download tool in China) in Sandboxie, the CPU usage of ekrn.exe was so high that I almost couldn't do anything, exclude restart my computer. But the bug disappeared after restart my computer.
> 
> https://forum.eset.com/topic/30272-ekrnexe-high-cpu-usage-50-due-to-context-switches-frequently/
> 
> See the reply on January 31.

This bug reproduce today, ekrn.exe cost>=50 CPU, and sandboxie couldn't delete contents, but after I try to unmount RAMDisk, this bug disappear immediately. The sandboxie work folder is storage in RAMDisk.

**Comment by mysteriously** [comment]
I don't use RAMdisk software. Did you try to apply a workaround? https://github.com/sandboxie-plus/Sandboxie/issues/1502#issuecomment-1036934666

**Comment by 0x391F** [comment]
> 

No. I fact, firefox disk cache has been disabled through set `browser.cache.disk.enable` to false and `browser.privatebrowsing.forceMediaMemoryCache` to true.
